### ISSUE TITLE
Faulty calculation of $base in protected function convertIntToBase62

### ISSUE THREAD

**Issue by thomashohn** [astehlik/typo3-extension-tinyurls#29]
Since there is no guarantee the the base62Dictionary has a length of 62 (if configured otherwise) that method should be changed from
```
  protected function convertIntToBase62(int $base10Integer, string $base62Dictionary): string
  {
      $base62Integer = '';
      $base = 62;

      do {
          $base62Integer = $base62Dictionary[$base10Integer % $base] . $base62Integer;
          $base10Integer = floor($base10Integer / $base);
      } while ($base10Integer > 0);

      return $base62Integer;
  }
```

to:

```
protected function convertIntToBase62(int $base10Integer, string $base62Dictionary): string
  {
      $base62Integer = '';
      // Use length of $base62Dictionary as base - stuff is a bit misleading if $base62Dictionary
      // doesn't contain 62 letters
      $base = strlen($base62Dictionary);

      do {
          $base62Integer = $base62Dictionary[($base10Integer % $base)] . $base62Integer;
          $base10Integer = floor($base10Integer / $base);
      } while ($base10Integer > 0);

      return $base62Integer;
  }
```

Or else the following error can be found in error logs
`PHP Warning: Uninitialized string offset 61 in web/typo3conf/ext/tinyurls/Classes/UrlKeyGenerator/Base62UrlKeyGenerator.php line 98`

**Comment by astehlik** [comment]
Thank you for taking the time to create an issue and you detailed description and your pull request.

I adjusted the behavior of the `Base62UrlKeyGenerator` based on your pull request.

Please let me know if the current develop branch works for you. If everything is OK I'll draft a new release.

Do you need this backported to older versions?

**Comment by thomashohn** [comment]
Hi - I ran it through a small test-program I created to test the output of the method. Seems to work fine, generates keys without throwing errors.

No backport needed since im on TYPO3 11 LTS
Thx. for improving it - you might want to rename the method convertIntToBase62 to convertIntToBaseX - but I believe this could be done when preparing tinyurls for TYPO3 12 LTS?

**Comment by astehlik** [comment]
Nice, thank you :+1: 

I just released versions [v11.1.0](https://github.com/astehlik/typo3-extension-tinyurls/releases/tag/v11.1.0) and  [v12.1.0](https://github.com/astehlik/typo3-extension-tinyurls/releases/tag/v12.1.0) containing the modifications.

Your right about the method naming but I did not want to break too much with this change. Basically the class would also have to be renamed. Maybe I'll change that in the future but for now I'm happy I got the TYPO3 12 version out the door :sweat_smile:
